openstack/networking/v2/extensions/provider/results.go

```diff
@@ -24,7 +19,7 @@ type NetworkProviderExt struct {
 	// segment depends on the segmentation model defined by network_type. For
 	// instance, if network_type is vlan, then this is a vlan identifier;
 	// otherwise, if network_type is gre, then this will be a gre key.
-	SegmentationID string `json:"-"`
+	SegmentationID int `json:"provider:segmentation_id"`
```

Comment:
This is not backward compatible, right?
@mandre @pierreprinetti what do you think?

Reply:
Correct, this isn't backward compatible. If we were to do this, it should be in its own PR for which we add the `breaking` label.

However, digging into this a bit, it seems the int vs string issue was previously discussed in #69 and we settled on the string type.

To quote @jtopjian :
> another argument for a string is in the case of zero value.
>
> Neutron is returning "null" rather than 0. Having a segmentation_id of 0 might cause [unexpected results](https://www.google.ca/?ion=1&espv=2#q=vlan%200).

Reply:
Yeah, there are three reasons a string might be used as the return/result type:

1. It truly is a string.
2. Duct tape to handle a wide array of return values (null vs 0).
3. Supporting multiple OpenStack versions and return types. I think the [hypervisors API extension](https://github.com/gophercloud/gophercloud/blob/master/openstack/compute/v2/extensions/hypervisors/results.go) is the winner for this. 🙂

Reply:
Hello @EmilienM , @mandre , @jtopjian 
I have updated PR to keep the `SegmentationID string` in `NetworkProviderExt` struct.
Although I am little bit confused we have string here but inside `Segment` struct there is `SegmentationID  int`. 

Hopefully now it all make sense together with the test :)

I would appreciate some review.
Thank you